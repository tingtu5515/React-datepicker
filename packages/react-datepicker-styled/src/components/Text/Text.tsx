import styled from 'styled-components'
import {
  fontFamily,
  FontFamilyProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps,
  color,
  ColorProps,
  lineHeight,
  LineHeightProps,
  space,
  SpaceProps,
} from 'styled-system'

interface TextProps
  extends FontWeightProps,
    FontSizeProps,
    FontFamilyProps,
    ColorProps,
    SpaceProps,
    LineHeightProps {}

const Text = styled('div')<TextProps>`
  ${fontFamily}
  ${fontSize}
  ${fontWeight}
  ${color}
  ${lineHeight}
  ${space}
`

export default Text
