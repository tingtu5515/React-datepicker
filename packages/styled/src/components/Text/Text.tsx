import styled from 'styled-components'
import {
  fontFamily,
  FontFamilyProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps,
  color,
  ColorProps,
  lineHeight,
  LineHeightProps,
  space,
  SpaceProps,
  compose,
} from 'styled-system'

interface TextProps
  extends FontWeightProps,
    FontSizeProps,
    FontFamilyProps,
    ColorProps,
    SpaceProps,
    LineHeightProps {}

const composeStyles = compose(
  fontFamily,
  fontSize,
  fontWeight,
  color,
  lineHeight,
  space,
)

const Text = styled('div')<TextProps>`
  ${composeStyles}
`

export default Text
