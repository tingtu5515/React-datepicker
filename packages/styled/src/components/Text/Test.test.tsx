import React from 'react'
import {render} from 'react-testing-library'
import Text from '.'

test('should match snapshot', () => {
  const {container} = render(
    <Text
      fontFamily="Montserrat"
      fontSize="14px"
      fontWeight={500}
      color="#000000"
      lineHeight={1.2}
      p="10px"
      m="10px"
    >
      Test
    </Text>,
  )
  expect(container).toMatchSnapshot()
  // @ts-ignore
  expect(container.firstChild).toHaveTextContent('Test')
})

test('should match style rules', () => {
  const {container} = render(
    <Text
      fontFamily="Montserrat"
      fontSize="14px"
      fontWeight={500}
      color="#000000"
      lineHeight={1.2}
      p="10px"
      m="10px"
    >
      Test
    </Text>,
  )
  // @ts-ignore
  expect(container.firstChild).toHaveStyleRule('font-family', 'Montserrat')
  // @ts-ignore
  expect(container.firstChild).toHaveStyleRule('font-weight', '500')
  // @ts-ignore
  expect(container.firstChild).toHaveStyleRule('font-size', '14px')
  // @ts-ignore
  expect(container.firstChild).toHaveStyleRule('color', '#000000')
  // @ts-ignore
  expect(container.firstChild).toHaveStyleRule('line-height', '1.2')
  // @ts-ignore
  expect(container.firstChild).toHaveStyleRule('margin', '10px')
  // @ts-ignore
  expect(container.firstChild).toHaveStyleRule('padding', '10px')
})
