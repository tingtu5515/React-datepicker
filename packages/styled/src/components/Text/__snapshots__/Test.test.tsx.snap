// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  font-family: Montserrat;
  font-size: 14px;
  font-weight: 500;
  color: #000000;
  line-height: 1.2;
  padding: 10px;
  margin: 10px;
}

<div>
  <div
    class="c0"
    color="#000000"
    font-family="Montserrat"
    font-size="14px"
    font-weight="500"
  >
    Test
  </div>
</div>
`;
